doc/SAI-Proposal-IPv6_Segment_Routing-1.md

```diff
@@ -0,0 +1,210 @@
+SAI IPv6 Segment Routing Proposal for SAI 1.2.0
+-------------------------------------------------------------------------------
+ Title       | SAI IPv6 Segment Routing
+-------------|-----------------------------------------------------------------
+ Authors     | Cavium Inc.
+ Status      | In review
+ Type        | Standards track
+ Created     | 04/14/2017
+ Updated     | 05/17/2017
+ SAI-Version | 1.2.0
+
+-------------------------------------------------------------------------------
+
+## Overview ##
+
+Segment Routing (SR) allows a node to steer a packet through a controlled set of instructions, called segments, by prepending an SR header to the packet.  A segment can represent any instruction, topological or service-based.  SR allows to enforce a flow through any path (topological, or application/service based) while maintaining per-flow state only at the ingress node to the SR domain.
+
+This document covers IPv6-based Segment Routing as per IETF Drafts:
+1. https://tools.ietf.org/html/draft-ietf-6man-segment-routing-header-06.  
+2. https://tools.ietf.org/html/draft-filsfils-spring-srv6-network-programming-00
+3. https://tools.ietf.org/html/draft-filsfils-spring-segment-routing-policy-00
+
+This specification proposes the following points:
+1. Introduce the concept of Segment Routing using IPv6.
+2. Introduce behavioral model modifications to support SR source, transit, and endpoint.
+3. Introduce SAI APIs to define SR properties including segment and TLV definitions.
+
+## Behavioral Model
+
+To clarify the behaviors of different nodes:
+1. Source Node.  This node first encapsulates the SR header on a native IPv6 packet and routes on the outer-most address in SR list.
+2. Transit Node.  Typically is unaware of SR header and only does native routing.  But there is an option to append the packet with another SR header to create a stack of SR headers via mechanism similar to Source Node.
```

Comment:
I find the following "transit-node" description quite precise. If you agree, perhaps we can adjust yours to avoid any misunderstandings.

As per SRv6-srh draft:

"...the only node who is allowed to inspect the Routing Extension Header (and therefore the SRH), is the node
   corresponding to the DA of the packet.  Any other transit node MUST NOT inspect the underneath routing header and MUST forward the packet towards the DA and according to the IPv6 routing table."